docs/questions/caching-in-chaincode.md

```diff
@@ -0,0 +1,54 @@
+### Question
+
+
+How do I implement caching in chaincode?
+
+
+### Answer
+
+
+Caching in GalaChain is automatically handled by the `GalaChainStubCache`. Here's what you need to know:
+
+1. Automatic Caching:
+   - The SDK handles caching automatically
+   - No manual cache implementation required
+   - Works within transaction boundaries
+   - Optimizes frequent reads of the same data
+
+2. Using the Cache:
+```typescript
+class GameContract extends Contract {
+  @Submit()
+  async processGameItems(
+    ctx: GalaChainContext,
+    params: { itemId: string }
+  ): Promise<void> {
+    // These calls automatically use the stub cache
+    const item = await getObjectByKey(ctx, GameItem, params.itemId);
+    
+    // Multiple reads of the same item use cache
+    const sameItem = await getObjectByKey(ctx, GameItem, params.itemId);
+    
+    // Range queries also use cache
+    const items = await getObjectsByPartialCompositeKey(
+      ctx,
+      GameItem.INDEX_KEY,
+      [item.owner]
+    );
+  }
+}
+```
+
+3. How It Works:
+   - Reads are cached in memory during transaction
+   - Writes are held in cache until transaction completion
+   - Cache is automatically flushed at transaction end
+   - All updates are written together
+
+4. Benefits:
+   - Improved performance for repeated reads
+   - Consistent view of state within transaction
+   - Reduced world state database load
+   - Automatic cache management
+
+Note: The stub cache is specifically designed for transaction-level caching. For cross-transaction data access patterns, consider implementing appropriate data structures and query patterns in your chaincode.
```

Comment:
also it's worth emphasizing that this is the different behavior that is supported by Fabric by default.

also we don't support cache for range reads (yet)